thumbor/engines/gif.py

```diff
@@ -118,8 +118,7 @@ def read(self, extension=None, quality=None):
 
         # Make sure gifsicle produced a valid gif.
         try:
-            with Image.open(BytesIO(self.buffer)) as image:
-                image.verify()
+            Image.open(BytesIO(self.buffer)).verify()
```

Comment:
Just make sure to call close() after verify.
The context manager was doing this: https://github.com/python-pillow/Pillow/blob/master/PIL/Image.py#L537